src/core/utils/ping_sender.cpp

```diff
@@ -108,6 +108,10 @@ Error PingSender::Ping(const Config &aConfig)
 
     mStatistics.Clear();
     mStatistics.mIsMulticast = static_cast<Ip6::Address *>(&mConfig.mDestination)->IsMulticast();
+    if (mStatistics.mIsMulticast)
```

Comment:
Should we avoid special-casing multicast?

Reply:
Updated. Note that multicast ping will wait for a full timeout for the final echo request, regardless of how many replies it has received.